types/shapediver/appbuilderinteractiontypes.ts

```diff
@@ -0,0 +1,106 @@
+export type InteractionParameterType = "selection" | "dragging" | "gumball";
+
+/**
+ * PART 1 - Parameter Definitions
+ * 
+ * These are the properties that are defined in GH.
+ * These definitions will in the future live in the Viewer.
+ */
+
+/**
+ * General properties of an interaction parameter.
+ */
+export interface IGeneralInteractionParameterProperties {
+    /** If the objects are hoverable. (default: true) */
+    hover?: boolean,
+    /** The names of the objects that can be interacted with. (see Jira document and discussion result) */
+    nameFilter?: string[]
+}
+
+/**
+ * Properties of a selection parameter.
+ */
+export interface IInteractionParameterPropsSelection extends IGeneralInteractionParameterProperties {
+    /** The minimum number of objects that can be selected. (default: 1) */
+    minimumSelection?: number,
+    /** The maximum number of objects that can be selected. (default: 1) */
+    maximumSelection?: number,
+}
+
+/**
+ * Properties of a dragging parameter.
+ * 
+ * This is a placeholder for now.
+ */
+export interface IInteractionParameterPropsDragging extends IGeneralInteractionParameterProperties {}
+
+/**
+ * Properties of a gumball parameter.
+ */
+export interface IInteractionParameterPropsGumball extends IGeneralInteractionParameterProperties {
```

Comment:
@MajorMeerkatThe3rd @snabela We also need a minimumSelection and maximumSelection here. Would be nice to have it in the parent interface but we agreed for now that the dragging input would only allow one element at a time.

Reply:
@mathieu-shapediver Makes sense, I forgot about that. Will be added within the next iteration.